extract.py

```diff
@@ -20,8 +20,16 @@ def main():
         tb = get_text_box_from_xml_element(textbox)
         for text_line in tb.text_lines:
             text_line.compact_texts()
+            t = ''
             for text in text_line.texts:
```

Comment:
Again, it seems like this is the natural thing to be iterating over, so I'd suggest making text_line iterable as well.